File: 04_linked_list/4_remove_nth_node_from_end.py
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        fast = head
        for _ in range(n):
            fast = fast.next
        if fast is None:
            return head.next
        slow = head
        while fast and fast.next:
            slow = slow.next
            fast = fast.next
        slow.next = slow.next.next
        return head

File: 04_linked_list/4_remove_nth_node_from_end.go
func removeNthFromEnd(head *ListNode, n int) *ListNode {

	fast := head
	for range n {
		fast = fast.Next
	}
	if fast == nil {
		return head.Next
	}
	
	slow := head
	for fast != nil && fast.Next != nil {
		slow = slow.Next
		fast = fast.Next
	}
	slow.Next = slow.Next.Next
	return head
}
